internal/dao/sqlx/reservation_dao.go

```diff
@@ -0,0 +1,72 @@
+package sqlx
+
+import (
+	"context"
+
+	"github.com/RHEnVision/provisioning-backend/internal/dao"
+	"github.com/RHEnVision/provisioning-backend/internal/db"
+	"github.com/RHEnVision/provisioning-backend/internal/models"
+	"github.com/jmoiron/sqlx"
+)
+
+const (
+	createReservation     = `INSERT INTO reservations (account_id, pubkey_id, status) VALUES ($1, $2, $3) RETURNING *`
+	deleteReservationById = `DELETE FROM reservations WHERE id = $1`
+)
+
+type reservationDaoSqlx struct {
+	name       string
+	create     *sqlx.Stmt
+	deleteById *sqlx.Stmt
+}
+
+func getReservationDao(ctx context.Context) (dao.ReservationDao, error) {
+	var err error
+	daoImpl := reservationDaoSqlx{}
+	daoImpl.name = "reservation"
+
+	daoImpl.create, err = db.DB.PreparexContext(ctx, createReservation)
+	if err != nil {
+		return nil, NewPrepareStatementError(ctx, &daoImpl, createPubkey, err)
+	}
+	daoImpl.deleteById, err = db.DB.PreparexContext(ctx, deleteReservationById)
+	if err != nil {
+		return nil, NewPrepareStatementError(ctx, &daoImpl, deletePubkeyById, err)
```

Comment:
Ditto. The reason why this can happen is that all the SQL queries are package-level constants and they are shared across multiple go source files.

Fixed.